integration-tests/tests/src/tests/notifications.ts

```diff
@@ -0,0 +1,463 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+class ListObject extends Realm.Object {
+  list!: Realm.List<TestObject>;
+  static schema = {
+    name: "ListObject",
+    properties: {
+      list: { type: "list", objectType: "TestObject" },
+    },
+  };
+}
+
+class TestObject extends Realm.Object {
+  doubleCol!: Realm.Types.Double;
+  static schema = {
+    name: "TestObject",
+    properties: {
+      doubleCol: "double",
+    },
+  };
+}
+
+describe("Notifications", () => {
+  let runCount = 0;
+
+  beforeEach(() => (runCount = 0));
+
+  describe("Realm notifications", () => {
+    openRealmBeforeEach({ schema: [TestObject] });
+
+    it("should fire on change", function (this: RealmContext, done) {
+      this.realm.addListener("change", (sender, event) => {
+        expect(sender).deep.equals(this.realm);
+        expect(event).equals("change");
+
+        const objects = this.realm.objects(TestObject);
+
+        expect(objects.length).equals(1);
+        expect(objects[0].doubleCol).equals(42);
```

Comment:
Kind of confused about the intention of these assertions (i'm sure this is just copied).  I think just making sure the callback gets called is enough of a test.

Reply:
Could also be later refactored